x/genaccounts/genesis_account.go

```diff
@@ -0,0 +1,125 @@
+package genaccounts
+
+import (
+	"errors"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
```

Comment:
Isn't this an anti-pattern? If this module is going to be tightly-bound to the `x/auth` module (and I expect it is), why not make it a submodule of `x/auth` instead?

Reply:
Def agree here. I think `x/genaccounts` is the right way to structure this. Like the submodule idea tho...

Reply:
I don't see this as an antipattern, I think it's totally fine to have modules which are tightly dependant on other modules (also those can eventually be abstracted with expected keepers to not be explicitly dependant on a particular module).

Also I'll mention that I don't think this should be a part of `auth` as eventually, we should be removing vesting accounts from `auth` (into a new module) and this module will depend on both... as per previous discussion this module should provide the general outline of what kinds of genaccount checks are required, however I think blockchain projects will often need to fork this for custom account fields -> meaning if this was in `auth` they would be required to fork auth which is something I see as undesirable (more difficult to upgrade or contribute back)

Reply:
If they require custom account fields shouldn't they be able to reimplement the `Account` interface?

Reply:
Also I'll mention that I actually used to hold the perspective that modules shouldn't import each other until I was convinced that this is fine by @jaekwon. The only thing is, we just ought to use caution when this is the case, and as possible abstract them out so they don't depend on each other of course.

Reply:
If we can abstract out some of the cross-dependencies I'm less against this, but I still think it is a bit odd to make it a module, since it doesn't have anything to do with processing transactions (or even helping other modules process transactions) - maybe we should create another category or another "GenesisModule" abstraction, something like that.

Reply:
Sure yeah I like the idea of having `GenesisOnlyModule` then `AppModule` can just extend that type as well! Also note that `x/` is not just "modules" per say it's kinda more like oddities which are often modules. but like also simulations isn't a module, but I think it belongs in `x/`